pkg/bmh/baremetalhost.go

```diff
@@ -526,8 +554,7 @@ func (builder *BmhBuilder) WaitUntilInStatus(status bmhv1alpha1.ProvisioningStat
 	}
 
 	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
-		var err error
-		builder.Object, err = builder.Get()
```

Comment:
I think this line is more accurate. You need to update builder.Object before checking .Status.Provisioning.State, otherwise you will just be reading the same value each time

Reply:
Or you can just use builder.Exists() and also return an error if it doesn't exist